From ee1ccbf056b392457540a5ffe184e193a1840e0d Mon Sep 17 00:00:00 2001
From: Ammar Zahid Ali Syed <ammar.zhd@googlemail.com>
Date: Mon, 7 Jul 2014 15:40:08 +0100
Subject: [PATCH 309/351] ASoC: florida: Added Support for Ez2Control 2.5A

Added firmware Id of Ez2Control 2.5A to
adsp2_irq function to enable ez2control
trigger for it as well.

Change-Id: Ibf9f11bd8a7aae22efc764823fe8f1ced34ed031
Signed-off-by: Ammar Zahid Ali Syed <ammar.zhd@googlemail.com>
---
 sound/soc/codecs/florida.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sound/soc/codecs/florida.c b/sound/soc/codecs/florida.c
index 40540cc..44802ab 100644
--- a/sound/soc/codecs/florida.c
+++ b/sound/soc/codecs/florida.c
@@ -1890,7 +1890,8 @@ static irqreturn_t adsp2_irq(int irq, void *data)
 	mutex_lock(&florida->compr_info.lock);
 
 	if (!florida->compr_info.trig &&
-	    florida->core.adsp[2].fw_id == 0x4000d &&
+	    (florida->core.adsp[2].fw_id == 0x4000d ||
+	    florida->core.adsp[2].fw_id == 0x40036) &&
 	    florida->core.adsp[2].running) {
 		if (florida->core.arizona->pdata.ez2ctrl_trigger)
 			florida->core.arizona->pdata.ez2ctrl_trigger();
-- 
1.7.9.5

